x-pack/metricbeat/module/azure/billing/service.go

```diff
@@ -85,46 +116,76 @@ func (service *UsageService) GetForecast(scope string, startTime, endTime time.T
 
 	// The aggregation is performed by the "sum" of "cost" for each day.
 	aggregationName := "Cost"
-	aggregationFunction := costmanagement.FunctionTypeSum
+	aggregationFunction := armcostmanagement.FunctionTypeSum
 
-	forecastDefinition := costmanagement.ForecastDefinition{
-		Dataset: &costmanagement.QueryDataset{
-			Aggregation: map[string]*costmanagement.QueryAggregation{
+	forecastDefinition := armcostmanagement.ForecastDefinition{
+		Dataset: &armcostmanagement.ForecastDataset{
+			Aggregation: map[string]*armcostmanagement.QueryAggregation{
 				"totalCost": {
-					Function: aggregationFunction,
+					Function: &aggregationFunction,
 					Name:     &aggregationName,
 				},
 			},
-			Granularity: costmanagement.GranularityTypeDaily,
+			Granularity: &armcostmanagement.PossibleGranularityTypeValues()[0],
```

Comment:
What about defining a local variable (like `aggregationName`) and taking the address of that? 

Maybe it's just me, but I feel uncomfortable with these `[N]`; you can't see the actual value unless you open the file where the slice is defined.

Same for the others `armcostmanagement.Possible*TypeValues()` occurrencies.

Reply:
I missed this one, in other parts of code I did it the way you suggested.